mobile/src/store/courses-management/reducer.ts

```diff
@@ -0,0 +1,48 @@
+import { createReducer } from '@reduxjs/toolkit';
+
+import { DataStatus } from '~/common/enums/enums';
+import { CourseGetResponseDto } from '~/common/types/types';
+
+import {
+  getCoursesWithCategory,
+  setNavigateFromCoursesManagement,
+  unsetNavigateFromCoursesManagement,
+} from './actions';
+
+type State = {
+  dataStatus: DataStatus;
+  coursesWithCategory: CourseGetResponseDto[];
+  totalCoursesNumber: number;
+  navigateFromCoursesManagement: boolean;
+};
+
+const initialState: State = {
+  dataStatus: DataStatus.IDLE,
+  coursesWithCategory: [],
+  totalCoursesNumber: 0,
+  navigateFromCoursesManagement: false,
```

Comment:
We don't need navigateFromCoursesManagement, and we also need categories because here we will have all categories and in courses screen only categories that have courses. Look how it's done on frontend side
```suggestion
  dataStatus: DataStatus.IDLE,
  courses: [],
  categories: [],
  totalCoursesNumber: 0,
```